csharp/src/SeedLang.Shell/Program.cs

```diff
@@ -13,20 +13,83 @@
 // limitations under the License.
 
 using System;
+using CommandLine;
+using CommandLine.Text;
 using SeedLang.Common;
 
 namespace SeedLang.Shell {
   class Program {
-    static void Main(string[] _) {
+    class Options {
+      [Option('r', "rule", Required = false, Default = "statement",
+        HelpText = "The parse rule of the source code." +
+                   " Allowed values are identifier, number, string, expression or statement.")]
+      public string Rule { get; set; }
+
+      [Option('l', "language", Required = false, Default = "python",
+        HelpText =
+          "The language of the source code. Allowed values are python, lua or javascript.")]
+      public string Language { get; set; }
+
+      [Option('t', "type", Required = false, Default = "source",
+        HelpText = "Run type. Allowed values are dryrun, source or bytecode.")]

```

Comment:
Optional: consider to auto-generate this type list from enum.GetName().

Reply:
The attribute argument must be constant expression. Keep it now, and see if I can find a better way later.